### PR TITLE
feat(components): [el-input-number] add focus & blur methods

### LINKED ISSUE #3334: [Bug Report] el-input-number did not trigger the focus and blur methods
### Element Plus version
1.1.0-beta.9

### OS/Browsers version
chrome93.0.4577.63

### Vue version
3.2.11

### Reproduction Link
https://codepen.io/pen/

### Steps to reproduce

```
  <el-input-number size="mini" ref="test" v-model="num4"></el-input-number>
    this.$refs.test.focus()
```

使用focus方法无法使焦点在该组件上,看了源码是没有写focus和blur方法

### What is Expected?
加入focus和blur方法

### What is actually happening?
focus和blur无效

### LINKED ISSUE #3361: [document update] [el-input-number] add focus & blur methods
This issue is to remind me that the documentation needs to be updated

> I was considering that we are redesigning the document, so do not be so anxious to modify.
> In addition, the translation of Japanese and French is too difficult.

Can we create an issue for that incase we forget the task? The documentation is still WIP(won't be long though)

_Originally posted by @JeremyWuuuuu in https://github.com/element-plus/element-plus/issues/3337#issuecomment-917614093_

### PR DESCRIPTION
**Pull request by adaex**
close #3334

- [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/input-number/src/index.vue [hunk 1]
```diff
@@ -249,6 +250,17 @@ export default defineComponent({
       }
       data.userInput = null
     }
+
+    const focus = () => {
+      nextTick(() => {
```
Comment (sxzz) [inline packages/components/input-number/src/index.vue:255]: Why use `nextTick`?

Comment (adaex) [reply packages/components/input-number/src/index.vue:255]: It was the reason for this bug at the time. https://github.com/ElemeFE/element/issues/18573

Comment (adaex) [reply packages/components/input-number/src/index.vue:255]: But this is `input-number`, it seems that this problem will never have. I will update it.

**Comment by sxzz** [comment]
Thanks. Could you please add this feature to docs?

**Review by sxzz — APPROVED** [review]

File: packages/components/input-number/src/index.vue [hunk 2]
```diff
@@ -54,6 +54,7 @@ import {
   defineComponent,
   reactive,
   ref,
+  nextTick,
```
Comment (sxzz) [inline packages/components/input-number/src/index.vue:57]: remove unused import please
```suggestion
```

**Comment by adaex** [comment]
I was considering that we are redesigning the document, so do not be so anxious to modify.
In addition, the translation of Japanese and French is too difficult.

**Comment by JeremyWuuuuu** [comment]
> I was considering that we are redesigning the document, so do not be so anxious to modify.
> In addition, the translation of Japanese and French is too difficult.

Can we create an issue for that incase we forget the task? The documentation is still WIP(won't be long though)

File: packages/components/input-number/src/index.vue [hunk 3]
```diff
@@ -249,6 +249,15 @@ export default defineComponent({
...
       data.userInput = null
     }
+
+    const focus = () => {
+      input.value.focus()
+    }
+
+    const blur = () => {
+      input.value.blur()
+    }
```
Comment (JeremyWuuuuu) [inline packages/components/input-number/src/index.vue:259]: Would be better with optional chaining though.
E.g.

```ts
input.value.blur?.()
```

**Review by jw-foss — APPROVED** [review]

File: packages/components/input-number/src/index.vue (hunk 3 above)
Comment (JeremyWuuuuu) [reply packages/components/input-number/src/index.vue:259]: This could help us with edge case

Comment (adaex) [reply packages/components/input-number/src/index.vue]: @JeremyWuuuuu  Yeah, that's a good idea.

**Comment by adaex** [comment]
> Can we create an issue for that incase we forget the task? The documentation is still WIP(won't be long though)

Yeah, I couldn't agree more. It's a really good idea.  

- #3361

**Review by sxzz — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
Great!!!! Let's roll it.

**Pull request merged**